transformer_engine/common/fused_softmax/scaled_masked_softmax_cuda.cu

```diff
@@ -305,6 +305,13 @@ __global__ void scaled_masked_softmax_warp_forward(
     }
     warp_reduce<acc_t, WARP_BATCH, WARP_SIZE, Max>(max_value);
 
+    // compute scale value to account for full mask
+    acc_t scale_value[WARP_BATCH];
```

Comment:
Neither this nor the max_value is assigned the default value. You need to be REALLY careful that the value gets assigned for all elements that are going to be read from (even for inputs that are potentially invalid), as otherwise this is an undefined behavior (and compiler is free to do anything in that case.

Reply:
Immediately after declaring `max_value` and `scale_value`, all elements in the array upto `WARP_BATCH` are populated, so this should be fine?

Reply:
Ok, I see that :+1: